rust/scx_utils/src/load_balance.rs

```diff
@@ -0,0 +1,318 @@
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+
+// This software may be used and distributed according to the terms of the
+// GNU General Public License version 2.
+
+//! # SCX Load Balancer
+//!
+//! A crate providing abstractions for implementing load balancing logic
+//! between scheduling domains.
+//!
+//! Initializing LoadBalancer
+//! -------------------------
+//!
+//! LoadBalancer objects are created from LoadAggregator objects. As the name
+//! suggests, LoadAggregator objects aggregate load passed by the caller, and
+//! create a LoadBalancer object that contains load and weights for each
+//! domain; adjusted for any infeasible weights.
+//!
+//! Note that the abstractions here are meant to be generic across schedulers.
+//! LoadAggregator assumes nothing about the scheduling domains being passed to
+//! it. For example, they may span layers defined in a scheduler, domains
+//! specified by the user via cpumask strings, or domains that span L3 caches.
+//! The crate simply aggregates load according to domain with IDs specified by
+//! the caller, computes global load sums, and adjusts any infeasible weights
+//! such that all infeasible scheduling entities still get their full duty
+//! cycle.
```

Comment:
The last sentence may be misleading for people who aren't already familiar with the problem. I think it'd be better to either give a bit longer explanation (ie. is infeasible too high or low? is weight being adjusted up or down?) or just direct to the document.